In [1]:
import pandas as pd
print("Success")

Success


In [2]:
url = "https://maven-datasets.s3.us-east-1.amazonaws.com/Data+Drills/SPY_close_price_5Y.csv"

In [5]:
spy_df = pd.read_csv(url, parse_dates=["Date"], index_col="Date")
spy_df.head()

,Close
Date,
2020-11-02,330.20
2020-11-03,336.03
2020-11-04,343.54
2020-11-05,350.24
2020-11-06,350.16


3 columns created: 50 days moving average, 200 days moving average, and the difference between the two moving averages.

In [9]:
spy_df.assign(
    ma_50 = lambda d: d.Close.rolling(window = 50, min_periods = 50).mean(),
    ma_200 = lambda d: d.Close.rolling(window = 200, min_periods = 200).mean(),
    # the 50 ma has to be grater than the 200 ma on the prior role the 50 ma has to be less than or equal the 200 ma to show where it changed
    golden_cross = lambda d: ((d.ma_50 > d.ma_200) & (d.ma_50.shift(1) <= d.ma_200.shift(1))).astype(int)
).query("golden_cross == 1")

,Close,ma_50,ma_200,golden_cross
Date,,,,
2023-02-02,416.78,394.3504,394.25025,1
2025-07-01,617.65,583.0978,582.04155,1


The code above calculates the 50-day and 200-day moving averages of the closing price of SPY, and identifies the dates where a "golden cross" occurs. A golden cross is a bullish signal that occurs when the 50-day moving average crosses above the 200-day moving average. The code then filters the DataFrame to show only the rows where a golden cross is identified. and the recent is 617.65. and that is the golden cross